src/lib/translators/mongo.ts

```diff
@@ -290,6 +290,53 @@ function transformReplace(step: Readonly<S.ReplaceStep>): MongoStep {
   };
 }
 
+/** transform an 'replace' step into corresponding mongo steps */
+function transformRollup(step: Readonly<S.RollupStep>): MongoStep {
+  const facet: { [id: string]: MongoStep[] } = {};
+  for (const [idx, elem] of step.hierarchy.entries()) {
+    const id = columnMap([...step.hierarchy.slice(idx), ...(step.groupby ?? [])]);
+    const aggs: { [id: string]: {} } = {};
+    for (const aggfStep of step.aggregations) {
+      if (aggfStep.aggfunction === 'count') {
+        aggs[aggfStep.newcolumn] = {
+          $sum: 1,
+        };
+      } else {
+        aggs[aggfStep.newcolumn] = {
+          [$$(aggfStep.aggfunction)]: $$(aggfStep.column),
+        };
+      }
+    }
+    const project: { [id: string]: string | number } = {
+      _id: 0,
+      ...Object.fromEntries(Object.keys(id).map(col => [col, `$_id.${col}`])),
+      ...Object.fromEntries(Object.keys(aggs).map(col => [col, 1])),
+      [step.labelCol ?? 'label']: `$_id.${elem}`,
+      [step.levelCol ?? 'level']: elem,
+    };
+    if (idx < step.hierarchy.length - 1) {
+      project[step.parentCol ?? 'parent'] = `$_id.${step.hierarchy[idx + 1]}`;
+    }
+    facet[`level_${idx}`] = [
```

Comment:
:exclamation: 
Usually, when describing a hierarchy, the order is from the root level to the leaves
level_0 -> root
level_1 -> children
level_2 -> grand children
and so on

We also generally observe this in tables:

continent | country | city
| -- | -- | -- |
europe | | |
europe | germany | |
europe | germany | berlin

It's easier to read and explore, ad if follows a standard "tree-like" pattern.

OK for you to reverse the order of the`hierarchy` parameter? 
It should also naturally set the order of the rows in the output table like above.

Reply:
(You'll have to modify the `id =` logic above)